components/fonts/platform/freetype/ohos/font_list.rs

```diff
@@ -81,11 +81,16 @@ struct FontList {
 fn enumerate_font_files() -> io::Result<Vec<PathBuf>> {
     let mut font_list = vec![];
     for elem in fs::read_dir(OHOS_FONTS_DIR)?.flatten() {
-        if elem.file_type().unwrap().is_file() {
+        if elem.file_type().is_ok_and(|file_type| file_type.is_file()) {
             let name = elem.file_name();
             let raw_name = name.as_bytes();
             if raw_name.ends_with(b".ttf".as_ref()) || raw_name.ends_with(b".ttc".as_ref()) {
-                debug!("Found font {}", elem.file_name().to_str().unwrap());
+                debug!(
+                    "Found font {}",
+                    elem.file_name()
+                        .to_str()
+                        .unwrap_or("<name containing non-utf8-data>")
```

Comment:
I'm not sure what the original error type is, but perhaps it contains the valid utf-8 substring up-until the original string was utf-8? in that case printing that could be helpful to pinpoint the error to a specific file.

Reply:
There's https://doc.rust-lang.org/std/str/struct.Utf8Error.html#method.valid_up_to but that doesn't seem worth the effort. I can use `String::from_utf8_lossy` though.